### DISCUSSION TITLE
Triton with Base Layer

### DISCUSSION THREAD

**Discussion by marcoma9023** [pelicanmapping/osgearth#2520] [Support Q&A]
HI,

I have a problem with Triton and using WMS layers at the same time.

Let me start by saying that I am working with OsgEarth 2.10.

I take the bathymetric data from GeoTiff, while the base layer is a WMS.

The problem lies in the fact that, what is known, is that with the three active layers the rendering is not fluid, while if I remove the wms layer at runtime, everything becomes fluid.

I tried not using the WMS but using the ArcGIS layer, but the problem is the same.

The callback does nothing.

With Wireshark I see that I make a lot of requests, but if I try to set the READ_ONLY property, the slowdown does not change.

```
osg::ref_ptr<osgEarth::Cache> cache = osgEarth::CacheFactory::create(options);
	osgEarth::Registry::instance()->setDefaultCache(cache);
	osgEarth::Registry::instance()->setDefaultCachePolicy(osgEarth::CachePolicy::USAGE_READ_WRITE);
	mapNode->getMap()->setCache(cache);
```

```
osgEarth::Triton::TritonOptions` tritonOptions;
	tritonOptions.user() = "ELMAN";
	tritonOptions.licenseCode() = "002021386e";
	tritonOptions.maxAltitude() = 500000;
	tritonOptions.useHeightMap() = true;
	tritonOptions.resourcePath() = _tritonResourcePath;
	_tritonCallback = new TritonCallback(_tritonDefaultProperties);
	_tritonLayer = new osgEarth::Triton::TritonLayer(tritonOptions, _tritonCallback);
	_tritonLayer->setName("triton");
	_tritonCallback->setTritonLayer(_tritonLayer);
	_tritonCallback->setViewer(_viewer);
	_tritonCallback->setWakeState(_showTritonWakes);
	_tritonCallback->setIntersectionState(_showTritonIntersections);
	_mapNode->getMap()->addLayer(_tritonLayer );
```

In the video I pan and zoom continuously and everything goes in jerks, while if I remove the layer, everything goes as smoothly as I would like and when I insert it again it goes back to being slow.

Do you have any advice or advice?

https://github.com/gwaldron/osgearth/assets/87537804/c5857271-ac75-4963-a8eb-9af0252c8177

**Comment by gwaldron** [comment]
Are you able to create an earth file we can use to reproduce the issue?
